dicoogle/src/main/java/pt/ua/dicoogle/server/users/UsersXML.java

```diff
@@ -164,96 +137,80 @@ public UsersStruct getXML()
             LoggerFactory.getLogger(UsersXML.class).error(ex.getMessage(), ex);
         }
-        
+
         Transformer serializer = hd.getTransformer();
         serializer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
         serializer.setOutputProperty(OutputKeys.METHOD, "xml");
         serializer.setOutputProperty(OutputKeys.INDENT, "yes");
-        serializer.setOutputProperty(OutputKeys.STANDALONE, "yes");   
-        try
-        {
+        serializer.setOutputProperty(OutputKeys.STANDALONE, "yes");
+        try {
             hd.setResult(streamResult);
             hd.startDocument();
-        } catch (SAXException ex)
-        {
+        } catch (SAXException ex) {
             LoggerFactory.getLogger(UsersXML.class).error(ex.getMessage(), ex);
         }
 
         AttributesImpl atts = new AttributesImpl();
-        try
-        {
+        try {
             //root element
             hd.startElement("", "", "Users", atts);
 
-            Iterator<User> us = UsersStruct.getInstance().getUsers().iterator();
+            Iterator<User> us = users.iterator();
 
             atts.clear();
-            while (us.hasNext())
-            {
+            while (us.hasNext()) {
                 User user = us.next();
-                
+
                 atts.addAttribute("", "", "username", "", user.getUsername());
-                atts.addAttribute("", "", "hash", "", user.getPasswordHash()) ;
+                atts.addAttribute("", "", "hash", "", user.getPasswordHash());
 
                 String temp = "false";
-                if(user.isAdmin())
+                if (user.isAdmin())
                     temp = "true";
-                if (user.getRoles()!=null&&user.getRoles().size()>0)
-                {
+                if (user.getRoles() != null && user.getRoles().size() > 0) {
                     String roles = "";
-                    for (Role r : user.getRoles())
-                    {
-                        roles+=r.getName()+",";
+                    for (Role r : user.getRoles()) {
+                        roles += r.getName() + ",";
                     }
                     StringUtils.removeEnd(roles, ",");
 
 
-                    atts.addAttribute("", "", "roles", "", roles ) ;
+                    atts.addAttribute("", "", "roles", "", roles);
                 }
 
-                atts.addAttribute("", "", "admin", "", temp) ;
+                atts.addAttribute("", "", "admin", "", temp);
 
                 hd.startElement("", "", "user", atts);
                 atts.clear();
                 hd.endElement("", "", "user");
             }
             hd.endElement("", "", "Users");
 
-
             hd.endDocument();
-        } catch (SAXException ex)
-        {
+        } catch (SAXException ex) {
             LoggerFactory.getLogger(UsersXML.class).error(ex.getMessage(), ex);
-        }
-        finally {
+        } finally {
             try {
                 out.close();
```

Comment:
Let's fix this while you're at it. Use the try-with resource syntax to close the printer, and make sure that it doesn't try to do something dubious if it fails (it should be a similar fix to #373).